lib/services/unified_resource.go

```diff
@@ -244,13 +244,13 @@ func (c *UnifiedResourceCache) iterateItems(ctx context.Context, start string, s
 						}
 					}
 
-					r, ok := cache.resources[item.Value]
+					collection, ok := cache.resources[item.Value]
 					if !ok {
 						return true
 					}
 
-					if len(kinds) == 0 || c.itemKindMatches(r, kindsMap) {
-						items = append(items, iteratedItem{key: item.Key, resource: r})
+					if len(kinds) == 0 || c.itemKindMatches(collection.getKind(), kindsMap) {
+						items = append(items, iteratedItem{key: item.Key, collection: collection.copyLocked()})
```

Comment:
it's quite expensive to copy here as the resource will be copied again outside after applying filter. any way to avoid it?